### ISSUE TITLE
Very strange Coding Problems

### ISSUE THREAD

**Issue by gmuehl** [ckeditor/ckeditor5-vue#106]
I have some very strange Problems with a Custom Build of CKEditor 5. Everything is fine when i serve the App with Node.js in Development Mode. As soon as i put a build on Apache I see this:

![image](https://user-images.githubusercontent.com/151238/69350359-b8400180-0c79-11ea-83fe-2cdc67d001da.png)

instead of nothing:

![Anmerkung 2019-11-21 160648](https://user-images.githubusercontent.com/151238/69350463-e3c2ec00-0c79-11ea-8e76-ed90ee4152bb.jpg)

i have no idea if it is a wrong encoded line-break or something like that. It replaces

`<p>&nbsp;</p>`

With a Paragraph and this strange A. It totally get's confused too, it says that the nodelist is damaged as soon as i start to edit. Now to the fun fact:

- It does not work with the classic-build
- It works fine with ckeditor4 and vue-ckeditor2

I found some really old issues that CKeditor4 once had the same problems, but it looks like it is working now. I have no clue what to check, except that it is working in any Editor except version 5 - so what could be the Problem?

**Comment by Mgsy** [comment]
Hi, just to be sure - did you follow our [Vue integration guide](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/vuejs.html)? Does this issue occur also in native CKEditor 5 (without any integration)? 

As I understood, you serve your application from Apache server and the editor's content is `<p>&nbsp;</p>` which results with strange `A` characters?

**Comment by gmuehl** [comment]
Exactly! I have tried

- Ckeditor 5 Custom Build
- Ckeditor 5 Classic Build
- Ckeditor 4

All in Vue. No problems with version 4, both builds of Ckeditor 5 showed a Â

If I should try 5 without vue please leave a notice.

**Comment by gmuehl** [comment]
I forgot the console errors. As soon as I type something it tells me that the nodelist is damaged. I can send you the complete error message as soon as I am home again.

**Comment by gmuehl** [comment]
Error message in Console, occurs when Editor gets the focus:

`Uncaught CKEditorError: model-nodelist-offset-out-of-bounds: Given offset cannot be found in the node list. Read more: https://ckeditor.com/docs/ckeditor5/latest/framework/guides/support/error-codes.html#error-model-nodelist-offset-out-of-bounds
 {"offset":2,"nodeList":[{"data":" "}]}`

**Comment by gmuehl** [comment]
Error does not occur in native CKEditor5 (without Vue, implementation follows quick start: https://ckeditor.com/docs/ckeditor5/latest/builds/guides/quick-start.html)

**Comment by Mgsy** [comment]
This is interesting issue. Can I ask you to provide exact steps to reproduce the issue?

**Comment by gmuehl** [comment]
That's a nice question, i have no Idea ;) All i can say at the moment is, that it is complicated ;) In a component i call the CKEditor

`<ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>`

editorData is loaded with Axios and is Part of a JSON-Object. The Object is generated with PHP and file_get_contents. So i could be possible, that Axios is the Problem, looks like it isn't PHP because it Works with the native CKEditor5.

Though, i added a Textarea to my View and injected {{editorData}} - No Â! I inject the same to ckEditor - Â shows up

Though, by now i found this

https://github.com/ckeditor/ckeditor5/issues/1431

And there are many old Posts regarding the Â in CKEditor4 from 2013

**Comment by gmuehl** [comment]
I start to think the Â are \n in the JSON-Response (newline).

**Comment by gmuehl** [comment]
This error is pretty strange, and it is definitely an Encoding Error. I am not sure who causes the Error, but if i paste the Axios Response in Vue, there are no Â - they just occur in the CKEDitor5 Textarea - i just upgraded from version 15 to 16. Though - i found out what causes the Error:

The HTML-Code is pretty old, and contains

`<p>&nbsp;</p>`

The blank Paragraph is shown, But somehow there is a Â before the & !

https://stackoverflow.com/questions/3597105/why-cant-i-get-rid-of-this-%C3%82nbsp

I tried to change the Encoding with PHP - without any luck. The {{editorData}} from Axios does not contain any Â, the CKEditor Textarea shows a Â for every <p>&nbsp;</p>

**Comment by FilipTokarski** [comment]
Hi!
I tried to reproduce this issue with custom CKEditor5 & Vue. I was using `<p>&nbsp;</p>` in `editorData` and I was also fetching it with Axios. I did it both in development mode and on Apache server and I was not able to get those strange 'Â' letters.
Could you please provide us with some code sample? I guess we need more precise information to reproduce this.

**Comment by gmuehl** [comment]
I reproduced the error with a new file, the Editor of my Customer encodes it in ANSI! I will attach the file to this comment, CKEditor shows a Â for each & in this file.

[impress.zip](https://github.com/ckeditor/ckeditor5-vue/files/4044628/impress.zip)

The workflow is pretty easy:

` <ckeditor :editor="editor" v-model="editorData" :config="editorConfig"></ckeditor>`

editorData is loaded with AXIOS and calls a PHP Script

```
.then(function (response) {
 self.editorData = response.data
})

```
And the PHP simply loads the file and JSON encodes it

```
$result = file_get_contents($filename);
echo json_encode($result);

```
I tried to change the Encoding from ANSI to UTF-8 with PHP:

` $result = mb_convert_encoding($result, "Windows-1252", "UTF-8");`

Which simply makes no difference, the Â will remain

ATTENTION: there will be no Â if you run `yarn run serve` - the Â is only shown in a real server environment without NODE.
